pyfr/partitioners/base.py

```diff
@@ -153,23 +173,49 @@ def _renumber_verts(self, mesh, vetimap, vparts):
         bndeti = set()
 
         # Identify vertices whose edges cross partition boundaries
-        for l, r in zip(*mesh['con_p0'][['f0', 'f1']].astype('U4,i4')):
+        for l, r in zip(*mesh['con_p0'][['f0', 'f1']].tolist()):
             l, r = tuple(l), tuple(r)
 
             if vpartmap[l] != vpartmap[r]:
                 bndeti |= {l, r}
 
-        # Move boundary vertices to the front of the list
+        # Move these exterior vertices to the start of the list
         nvetimap, nvparts = list(bndeti), [vpartmap[eti] for eti in bndeti]
 
-        # Followed by the internal vertices
-        for eti, part in zip(vetimap, vparts):
-            if eti not in bndeti:
-                nvetimap.append(eti)
-                nvparts.append(part)
+        # Next, process the internal vertices
+        for k, v in mesh.items():
+            if k.startswith('spt'):
+                etype = k.split('_')[1]
+
+                # Start with curved elements, followed by linear elements
```

Comment:
Are curved elements in the exterior taken into account somewhere?

Reply:
The plan is to treat all exterior elements as if they were curved.  (Although as discussed offline we're probably not going to make use of internal/external in the solver, but will keep it in the file.)